qiskit/algorithms/optimizers/gradient_descent.py

```diff
@@ -149,60 +227,99 @@ def settings(self) -> Dict[str, Any]:
             "callback": self.callback,
         }
 
-    def minimize(
+    def ask(self) -> AskObject:
+        """
+        For gradient descent this method simply returns an
+        :class:`qiskit.algorithms.optimizers.AskObject` containing the current point for the
+        gradient to be evaluated, a parameter epsilon in case the gradient has to be approximated
+        and a bool flag telling the prefered way of evaluating the gradient.
+        """
+        return AskObject(
+            x_jac=self._state.x,
+        )
+
+    def tell(self, ask_object: AskObject, tell_object: TellObject) -> None:
+        """
```

Comment:
Methods etc here al lshould have the standard docstring format we use (Google) and have Args, Returns etc. This is one example I chose to comment on - others here need similar corrections